test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/GraphNodeTests.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using FluentAssertions;
+using NuGet.LibraryModel;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.DependencyResolver.Core.Tests
+{
+    public class GraphNodeTests
+    {
+        [Theory]
+        [InlineData(null, true)]
+        [InlineData(false, false)]
+        [InlineData(true, true)]
+        public void GraphNode_WithDifferentHasEmptyParentNodes_CreateParentNodesCorrectly(bool? hasParentNodes, bool parentNodesHasNode)
+        {
+            LibraryRange libraryRangeA = GetLibraryRange("A", "1.0.0", "1.0.0");
+            LibraryRange libraryRangeB = GetLibraryRange("B", "1.0.0", "1.0.0");
+
+            GraphNode<RemoteResolveResult> nodeA, nodeB;
+
+            if (!hasParentNodes.HasValue)
+            {
+                nodeA = new GraphNode<RemoteResolveResult>(libraryRangeA);
+                nodeB = new GraphNode<RemoteResolveResult>(libraryRangeB);
+            }
+            else
+            {
+                nodeA = new GraphNode<RemoteResolveResult>(libraryRangeA, hasInnerNodes:true, hasParentNodes.Value);
+                nodeB = new GraphNode<RemoteResolveResult>(libraryRangeB, hasInnerNodes:true, hasParentNodes.Value);
+            }
+
+            bool sameParentNodes = (nodeA.ParentNodes == nodeB.ParentNodes);
```

Comment:
I feel it's more intuitive, test exactly what is value it should expect.
`bool sameParentNodes = (nodeA.ParentNodes == emptyNode) && (nodeB.ParentNodes == emptyNodes);`